File: 프로그래머스/lv1/크기가 작은 부분 문자열.py
def solution(t, p):
    answer = 0
    num_list=[]
    for i in range(len(t)-len(p)+1):
        num_list.append(t[i:i+len(p)])
    for i in num_list:
        if int(i)<=int(p):
            answer+=1
    return answer

print(solution("3141592","271"))

File: 프로그래머스/lv1/크기가작은부분문자열.java
package lv1;

public class 크기가작은부분문자열 {
    public int solution(String t, String p) {
        int answer = 0;
        long b=Long.parseLong(p);
        for(int i=0;i<=t.length()-p.length();i++){
            long a=Long.parseLong(t.substring(i,i+p.length()));
            if(a<=b){
                answer++;
            }
        }
        return answer;
    }

    public static void main(String[] args) {
        크기가작은부분문자열 smallString = new 크기가작은부분문자열();
        System.out.println(smallString.solution("3141592","271"));
    }
}
